src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -78,7 +78,11 @@ protected function createController($controller)
      */
     protected function instantiateController($class)
     {
-        $controller = parent::instantiateController($class);
+        if ($this->container->has($class)) {
+            $controller = $this->container->get($class);
```

Comment:
the simple way to fix my comment about ContainerAwareInterface is to return directly here